### DISCUSSION TITLE
Can I have the target rate sensor re-evaluate multiple times a day - every 30 minutes?

### DISCUSSION THREAD

**Discussion by rcourtman** [BottlecapDave/HomeAssistant-OctopusEnergy#851] [Q&A]
I apologise if this has been asked already, but I did have a look and I didn't find anything that answered this specific question. If I'm on Agile and I'm looking at the best three (1.5) time slots for instance, and I have the reevaluate ticked, I was hoping after the first slot had passed, it could re-evaluate the next best three. 

As I'm writing this, I can see how it's somewhat flawed because this would be solved if I was to increase the 'hours you require in decimal format' from 1.5 to something bigger. The problem is i need to set my target sensor to work in three slots only because my solis inverter only allows me to program in three time slots. 

I appreciate there are other ways I could work around this however my setup which i've spent quite a long time to get to where it is.. uses a target rate sensor with three slots only.

**Comment by BottlecapDave** [comment]
Hello. New slots are calculated once all initial slots have passed. This is to keep processing down and the fact the sensors main purpose is to find slots within a 24 hour period, not to be continual. This is to ensure it comes on for the specified amount of time and not go over or only partially activate.

You could possibly work around this by calling the update_config service with the same hours. This will reset the selected target rate times and force the sensor to re-evaluate them. Please be aware that the time period it will look at will gradually get smaller until it goes into the new period (e.g. if the sensor is set to look between 16:00 and 16:00, then at 12:00 it will only look at rates between 12:00 and 16:00), meaning that the selected times will gradually get less efficient.
